cmd/main.go

```diff
@@ -355,3 +361,21 @@ func autoMigrationWithLeader(migrationLeader leader.ElectorInterface, db *gorm.D
 		return err
 	})
 }
+
+func uploadBaseISOWithLeader(uploadLeader leader.ElectorInterface, objectHandler s3wrapper.API, generator generator.ISOInstallConfigGenerator, log logrus.FieldLogger) error {
+	ctx := context.Background()
+	return uploadLeader.RunWithLeader(ctx, func() error {
```

Comment:
Not sure i understand the flow, is that a blocking function? does it wait to become a leader?

Reply:
yap, it will wait till it become leader

Reply:
Yes, it waits to become a leader and runs the job.  The job also checks if the image exists and skips the upload if it does.

Reply:
OK thanks.